common/winobjc.versioning.common.targets

```diff
@@ -0,0 +1,245 @@
+
+      <PackageResourceVersion>$(PackageVersion_Major).$(PackageVersion_Minor).$(PackageResourceVersion_Build).$(PackageResourceVersion_Revision)</PackageResourceVersion>
+    </PropertyGroup>
+  </Target>
+
+  <!--
+    VCXPROJ Versioning target
+
+    This target will inject a common resource file (common.rc in this same directory) into the compilation of any .dll / .exe that imports
+    this .targets file. Preprocessor definitions are used to keep the verioning dynamic and configurable.
+  -->
+  <Target Name="AddCommonRCFile" 
+          BeforeTargets="BeforeResourceCompile" 
+          DependsOnTargets="CalculatePackageVersion" 
+          Condition="'$(ProjectExt)' == '.vcxproj' And '$(ConfigurationType)' != 'StaticLibrary'">
+
+    <PropertyGroup>
+      <PackageResourceVersion_Alt>$(PackageVersion_Major),$(PackageVersion_Minor),$(PackageResourceVersion_Build),$(PackageResourceVersion_Revision)</PackageResourceVersion_Alt>
+    </PropertyGroup>
+
+    <ItemGroup>
+      <ResourceCompile Include="$(MSBuildThisFileDirectory)\common.rc">
+        <PreprocessorDefinitions>%(PreprocessorDefinitions);GIT_VERSION_ALT=$(PackageResourceVersion_Alt);GIT_VERSION=$(PackageResourceVersion);TARGET_NAME=$(TargetFileName);PACKAGE_VERSION=$(PackageVersion)</PreprocessorDefinitions>
+        <ResourceOutputFileName>$(TargetDir)$(TargetName).res</ResourceOutputFileName>
+      </ResourceCompile>
+    </ItemGroup>
+  </Target>
+
+
+    <!--
+    VSIX Versioning target
+
+    This target will modify the VSIX manifest that is placed in the intermediate directory during the compilation of a vsix.
+    By using the intermediate directory, the working directory state isn't sullied with generated / temporary data. This does
+    mean though the fixup target needs to operate more "in the build" so it isn't missed by any other vsix step. This is why
+    several "DependsOn" properties now also include this target.
+  -->
+  <PropertyGroup>
+    <GetVsixDeploymentPathDependsOn>$(GetVsixDeploymentPathDependsOn);FixUpIntermediateVSIXManifest</GetVsixDeploymentPathDependsOn>
+    <CopyVsixManifestFileDependsOn>$(CopyVsixManifestFileDependsOn);FixUpIntermediateVSIXManifest</CopyVsixManifestFileDependsOn>
+    <GetVsixSourceItemsDependsOn>$(GetVsixSourceItemsDependsOn);FixUpIntermediateVSIXManifest</GetVsixSourceItemsDependsOn>
+  </PropertyGroup>
+
+  <Target Name="FixUpIntermediateVSIXManifest"
+          BeforeTargets="CreateVsixContainer"
+          DependsOnTargets="CalculatePackageVersion;DetokenizeVsixManifestFile"
+          Condition="'$(CreateVsixContainer)' == 'true'">
+
+    <XmlPoke XmlInputPath="$(IntermediateVsixManifest)"
+             Namespaces="&lt;Namespace Prefix='vsix' Uri='http://schemas.microsoft.com/developer/vsx-schema/2011' /&gt;"
+             Query="/vsix:PackageManifest/vsix:Metadata/vsix:Identity/@Version"
+             Value="$(PackageResourceVersion)"/>
+  </Target>
+
+
+  <!--
```

Comment:
Will this overwrite whatever AssemblyInfo.cs file is already included in a csproj? Or just add on to it?